src/enum_net.cpp

```diff
@@ -1185,18 +1193,26 @@ int _System __libsocket_sysctl(int* mib, u_int namelen, void *oldp, size_t *oldl
 			return std::vector<ip_route>();
 		}
 
-		std::uint32_t seq = 0;
+		struct
+		{
+			struct nlmsghdr hdr;
+			struct rtmsg msg;
+		} request{};
+		request.hdr.nlmsg_len = std::uint32_t(NLMSG_LENGTH(sizeof(request.msg)));
+		request.msg.rtm_family = AF_UNSPEC;
```

Comment:
moving the request out of the `nl_dump_request()` function also offers some additional flexibility, in case we want to specify which routing table we want to enumerate for instance, or setting other fields in `rtmsg`